### PR TITLE
add liquidity manager

### PR DESCRIPTION
**Pull request by gregLibert**

### CODE REVIEW

File: assembly/helpers/liquidityManager.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+  }
+
+  /**
+   * Retrieves the balance for a specified user address.
+   *
+   * @param user - The user's address.
+   *
+   * @returns User's balance, or 0 if no balance is found.
+   */
+  public getBalance(user: Address): T {
```
Comment (leoloco) [inline assembly/helpers/liquidityManager.ts:95]: People are used to `balanceOf` bu it doesn't make such a difference

File: assembly/helpers/liquidityManager.ts [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+   * If the spender has no allowance in the storage, a new allowance is created.
+   *
+   * @param owner - Owner's address.
+   * @param spender - Spender's address.
+   * @param deltaAmount - Amount to adjust the allowance by.
+   * @param increase - Whether to increase (true) or decrease (false) the allowance.
+   *
+   * @throws if attempting to decrease allowance below available amount.
+   */
+  public updateAllowance(
```
[the remark this answers is not in the document]
Comment (gregLibert) [reply assembly/helpers/liquidityManager.ts:218]: The delta logic is key as it solves the [multiple withdrawal attack](https://www.researchgate.net/publication/335362808_Resolving_the_Multiple_Withdrawal_Attack_on_ERC20_Tokens).

**Comment by gregLibert** [comment]
> If I'd have to make some remarks on this particular object, I'd say we could try to match the standard ERC20 naming, i.e., getAllowance -> allowance, getBalance -> balanceOf, updateAllowance -> approve.

We shouldn't mix smart contract exported functions and object methods. I don't intend to follow any ERC/EIP on objects. I prefer to focus on coding best practices. (Duplicate comment)

File: assembly/helpers/liquidityManager.ts (hunk 1 above)
Comment (leoloco) [reply assembly/helpers/liquidityManager.ts:95]: Understood, but then, following this logic, shouldn't we rename `mint` and `burn` with `issue` and `destroy` ?

**Review by Ben-Rey — APPROVED** [review]

File: assembly/helpers/liquidityManager.ts (hunk 1 above)
Comment (gregLibert) [reply assembly/helpers/liquidityManager.ts:95]: Because I like mint and burn. I find it explicit.

**Review by thomas-senechal — APPROVED** [review]

File: assembly/helpers/liquidityManager.ts (hunk 1 above)
Comment (leoloco) [reply assembly/helpers/liquidityManager.ts:95]: But then you are potentially mixing up smart contract exported functions and object methods

**Pull request merged**